include/crocoddyl/multibody/wrench-cone.hxx

```diff
@@ -0,0 +1,140 @@
+// Copyright note valid unless otherwise stated in individual files.
+// All rights reserved.
+///////////////////////////////////////////////////////////////////////////////
+
+#include <math.h>
+#include <iostream>
+
+namespace crocoddyl {
+
+template <typename Scalar>
+WrenchConeTpl<Scalar>::WrenchConeTpl() : nf_(16) {
+  A_.resize(nf_ + 1, 6);
+  ub_.resize(nf_ + 1);
+  lb_.resize(nf_ + 1);
+  // compute the matrix
+  update(Matrix3s::Identity(), Scalar(0.7), Vector2s(0.1, 0.05));
+}
+
+template <typename Scalar>
+WrenchConeTpl<Scalar>::WrenchConeTpl(const Matrix3s& rot, const Scalar& mu, const Vector2s& box_size, std::size_t nf)
+    : nf_(nf) {
+  if (nf_ % 2 != 0) {
+    nf_ = 16;
+    std::cerr << "Warning: nf has to be an even number, set to 16" << std::endl;
+  }
+  A_.resize(nf_ + 1, 6);
+  ub_.resize(nf_ + 1);
+  lb_.resize(nf_ + 1);
+  // compute the matrix
+  update(rot, mu, box_size);
+}
+
+template <typename Scalar>
+WrenchConeTpl<Scalar>::WrenchConeTpl(const WrenchConeTpl<Scalar>& cone)
+    : A_(cone.get_A()),
+      ub_(cone.get_ub()),
+      lb_(cone.get_lb()),
+      rot_(cone.get_rot()),
+      box_(cone.get_box()),
+      mu_(cone.get_mu()),
+      nf_(cone.get_nf()) {}
+
+template <typename Scalar>
+WrenchConeTpl<Scalar>::~WrenchConeTpl() {}
+
+template <typename Scalar>
+void WrenchConeTpl<Scalar>::update(const Matrix3s& rot, const Scalar& mu, const Vector2s& box_size) {
+  rot_ = rot;
+  mu_ = mu;
+  box_ = box_size;
+  
+  mu_ = mu_ / sqrt(Scalar(2.));
+  A_.setZero();
+
+  // Wrench friction cone (inertial frame)
+  //
+  //     A_ * wrench <= 0
+  //
+  // where ``wrench`` is expressed in the inertial frame located at the
+  // center of the rectangular foot contact area (X, Y), with axes parallel to
+  // those of the world frame.
+
+  A_.row(0).head(3) << 1., 0., -mu_;
+  A_.row(1).head(3) << -1., 0., -mu_;
+  A_.row(2).head(3) << 0., 1., -mu_;
+  A_.row(3).head(3) << 0., -1., -mu_;
+  A_.row(4).head(3) << 0., 0., -1.;
+  A_.row(5).segment(2, 3) << -box_(1), 1., 0.;
+  A_.row(6).segment(2, 3) << -box_(1), -1., 0.;
+  A_.row(7).segment(2, 3) << -box_(0), 0., 1.;
+  A_.row(8).segment(2, 3) << -box_(0), 0., -1.;
+  A_.row(9) << box_(1), box_(0), -mu_ * (box_(0) + box_(1)), -mu_, -mu_, -1.;
+  A_.row(10) << box_(1), -box_(0), -mu_ * (box_(0) + box_(1)), -mu_, mu_, -1.;
+  A_.row(11) << -box_(1), box_(0), -mu_ * (box_(0) + box_(1)), mu_, -mu_, -1.;
+  A_.row(12) << -box_(1), -box_(0), -mu_ * (box_(0) + box_(1)), mu_, mu_, -1.;
+  A_.row(13) << box_(1), box_(0), -mu_ * (box_(0) + box_(1)), mu_, mu_, 1.;
+  A_.row(14) << box_(1), -box_(0), -mu_ * (box_(0) + box_(1)), mu_, -mu_, 1.;
+  A_.row(15) << -box_(1), box_(0), -mu_ * (box_(0) + box_(1)), -mu_, mu_, 1.;
+  A_.row(16) << -box_(1), -box_(0), -mu_ * (box_(0) + box_(1)), -mu_, -mu_, 1.;
+  
+  Matrix6s c_R_o = Matrix6s::Zero();
+  c_R_o.topLeftCorner(3,3) = rot_;
+  c_R_o.bottomRightCorner(3,3) = rot_;
+  
+  MatrixX6s A_local = A_;
+  A_ = A_local * c_R_o;
```

Comment:
There is no need to create a local data, however, this comment is not strictly important (it is more instructive).
You could use `eval()` command in Eigen to do these operations. I invite you to read more about this in Eigen documentation.

For this case you would do something like:
```cpp
A_ = (A_local * c_R_o).eval();
```